pcb/rules/rule9_2.py

```diff
@@ -20,22 +20,24 @@ def check(self):
         # check if module is through-hole
         if module.attribute == 'pth':
             pads = module.getPadsByNumber(1)
-            if len(pads)==0:
+            if len(pads) == 0:
                 pads = module.getPadsByNumber('A1')
-            self.pin1_count = len(pads)
-            self.pin1_position = []
-
-            # check/get pads positions
-            anchor_ok = False
+                
+            if len(pads) == 0:
+                self.error("Pad 1 not found in footprint")
+                return True
+                
```

Comment:
The two varoables `self.pin1_count` and `self.pin1_position` are no longer set: 

```python
self.pin1_count=len(pads)
            if len(pads) == 0:
                self.error("Pad 1 not found in footprint")
                return True
                
            for pad in pads:
                pos = pad['pos']
                if len(self.pin1_position)==0: 
                    self.pin1_position=[pos['x'],pos['y']]
```

![2017-03-12 17_53_45-pcb_rules_rule9_2 py_ working tree - tortoisegitmerge](https://cloud.githubusercontent.com/assets/11967523/23833891/0c7deba2-074d-11e7-96cd-d86c723ca8cc.png)